pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -735,27 +794,62 @@ installPhase:
 
 		default:
 			for _, installWorker := range pplnPhase.q {
+				if enterpriseApi.ScopeLocal != getAppSrcScope(installWorker.afwConfig, installWorker.appSrcName) {
+					scopedLog.Error(nil, "Install worker with wrong scope", "name", installWorker.cr.GetName(), "namespace", installWorker.cr.GetNamespace(), "pod name", installWorker.targetPodName, "App name", installWorker.appDeployInfo.AppName, "digest", installWorker.appDeployInfo.ObjectHash)
+					continue
+				}
+
 				phaseInfo := getPhaseInfoByPhaseType(installWorker, enterpriseApi.PhaseInstall)
 				if checkIfWorkerIsEligibleForRun(installWorker, phaseInfo, enterpriseApi.AppPkgInstallComplete) {
 					installWorker.waiter = &pplnPhase.workerWaiter
 					select {
 					case pplnPhase.msgChannel <- installWorker:
 						scopedLog.Info("Install worker got a run slot", "name", installWorker.cr.GetName(), "namespace", installWorker.cr.GetNamespace(), "pod name", installWorker.targetPodName, "App name", installWorker.appDeployInfo.AppName, "digest", installWorker.appDeployInfo.ObjectHash)
 						installWorker.isActive = true
+
+						// ToDo Start: sgontla: Fix the flow once CSPL-1449 is in place
+						var localInstallCtxt *localScopeInstallContext = &localScopeInstallContext{
+							worker: installWorker,
+							//sem:
+						}
+
+						installWorker.waiter.Add(1)
+						localInstallCtxt.runPlaybook()
+						// just drain the channel for now, should handle in CSPL-1449
+						//<-pplnPhase.msgChannel
+						// ToDo End: sgontla: Fix the flow once CSPL-1449 is in place
 					default:
 						installWorker.waiter = nil
 					}
 				} else if phaseInfo.Status == enterpriseApi.AppPkgInstallComplete {
-					if installWorker.cr.GetObjectKind().GroupVersionKind().Kind == "Standalone" && isAppInstallationCompleteOnStandaloneReplicas(installWorker.appDeployInfo.AuxPhaseInfo) {
-						installWorker.appDeployInfo.PhaseInfo.Phase = enterpriseApi.PhaseInstall
-						installWorker.appDeployInfo.PhaseInfo.Status = enterpriseApi.AppPkgInstallComplete
+					var needToDeleteAppFromOperator bool = false
+					if installWorker.cr.GetObjectKind().GroupVersionKind().Kind == "Standalone" {
+						if installWorker.sts != nil && *installWorker.sts.Spec.Replicas > 1 && isAppInstallationCompleteOnStandaloneReplicas(installWorker.appDeployInfo.AuxPhaseInfo) {
+							installWorker.appDeployInfo.PhaseInfo.Phase = enterpriseApi.PhaseInstall
+							installWorker.appDeployInfo.PhaseInfo.Status = enterpriseApi.AppPkgInstallComplete
+							needToDeleteAppFromOperator = true
+						}
+					} else {
+						needToDeleteAppFromOperator = true
+					}
+
+					if needToDeleteAppFromOperator {
 						installWorker.appDeployInfo.DeployStatus = enterpriseApi.DeployStatusComplete
+						appPkgLocalPath := getAppPackageLocalPath(installWorker)
+						err := os.Remove(appPkgLocalPath)
```

Comment:
you would also need to free up space in terms of `availableDiskSpace` which the downloads threads check